### ISSUE TITLE
 FATAL an application with the same name and a different base already exists, consider choosing a different name

### ISSUE THREAD

**Issue by hookenful** [argoproj-labs/argocd-autopilot#470]
Hi there!

I am trying to create the same app for different projects (environments)

`argocd-autopilot application create frontend --app http://192.168.0.78:8080/root/machairodus-new/web/kustomize/envs/dev?ref=1.5 --project dev--dest-namespace dev`

`argocd-autopilot application create frontend --app http://192.168.0.78:8080/root/machairodus-new/web/kustomize/envs/test?ref=1.5 --project test --dest-namespace test`

In result I have an error:

```
FATAL an application with the same name and a different base already exists, consider choosing a different name
```

How should I specify different kustomize manifests for different envs? Is there any working example to achieve that with `argocd-autopilot`

**Comment by neeraj-ec** [comment]
I think this happens when you have `kustomization` files in the same git repo which is used by `argocd-autopilot`. For example you created your app yaml files in `$GIT_REPO/apps/sample-app/overlayes/dev`. Here $GIT_REPO is same git repo you used for `export GIT_REPO=https://....`

I'm also struggling with the same issue. The documentation about this project poor and skips a lot of details and too hard to follow for new gitops user who does not have much knowledge about argocd.

@noam-codefresh Is it mandatory to keep your application yaml files somewhere else other than `$GIT_REPO/apps/app1/overlays/project1`. If yes I don't understand how it achieve gitops then? If I'm putting my yaml files in some other dir or git repo then how argocd will reconcile? Because it is watching `$GIT_REPO/apps/app1/overlays/project1` directory for changes and any new deployment will make changes in yaml files which are at some other place.

---
My file structure is
```
.
├── apps/
│   └── app1/
│       ├── base/
│       │   ├── dep.yaml
│       │   ├── configs.yaml
│       │   ├── service.yaml
│       │   ├── ingress.yaml
│       │   └── kustomization.yaml
│       └── overlayes/
│           ├── dev/
│           │   └── kustomization.yaml
│           ├── prod/
│           │   └── kustomization.yaml
│           └── stage/
│               └── kustomization.yaml
├── bootstrap
└── projects
```

I create `apps/app1` directory manually, commit to $GIT_REPO and then run the command 
`argocd-autopilot app create app1 --app $GIT_REPO/apps/app1/overlays/dev/ --project dev --type kustomize --wait-timeout 2m`

And I get the error.
``` log
FATAL an application with the same name and a different base already exists, consider choosing a different name
```

@noam-codefresh @danielm-codefresh Can somebody please help me understand whats I'm doing wrong?

**Comment by hookenful** [comment]
@neeraj-ec I think u just need to keep your base app manifests outside gitops repository (e.g in app repository), but after u `argocd-autopilot app create ...`, you can manually add overlay where u specify params, especially image and etc.. So after that, u can modify that image section in gitops repository and it argocd will reconcile.

@noam-codefresh am I right with that?

**Comment by neeraj-ec** [comment]
> @neeraj-ec I think u just need to keep your base app manifests outside gitops repository (e.g in app repository), but after u `argocd-autopilot app create ...`, you can manually add overlay where u specify params, especially image and etc.. So after that, u can modify that image section in gitops repository and argocd will reconcile.
> 
> @noam-codefresh am I right with that?

That is one way of doing it. But I want to keep all my manifests files in a single `infra` repository. Managing whole k8s cluster from single repo. In that case, https://github.com/argoproj-labs/argocd-autopilot/issues/470#issuecomment-1557178074 works fine.
